addons/main.go

```diff
@@ -303,12 +303,29 @@ func enableClusterBootstrapAndConfigControllers(ctx context.Context, mgr ctrl.Ma
 }
 
 func enableWebhooks(ctx context.Context, mgr ctrl.Manager, flags *addonFlags) {
+	managementFrequency := time.Second * 60 // This is how often we manage the webhook tls. This is how often we check certificates and corret if invalid.
 	certPath := path.Join(constants.WebhookCertDir, "tls.crt")
 	keyPath := path.Join(constants.WebhookCertDir, "tls.key")
 	if _, err := webhooks.InstallNewCertificates(ctx, mgr.GetConfig(), certPath, keyPath, constants.WebhookScrtName, flags.addonNamespace, constants.WebhookServiceName, constants.AddonWebhookLabelKey+"="+constants.AddonWebhookLabelValue); err != nil {
 		setupLog.Error(err, "unable to install certificates for webhooks")
 		os.Exit(1)
 	}
+	webhookTLS := webhooks.WebhookTLS{
+		Ctx:           ctx,
+		K8sConfig:     mgr.GetConfig(),
+		CertPath:      certPath,
+		KeyPath:       keyPath,
+		Name:          constants.WebhookScrtName,
+		ServiceName:   constants.WebhookServiceName,
+		LabelSelector: constants.AddonWebhookLabelKey + "=" + constants.AddonWebhookLabelValue,
+		Logger:        setupLog,
+		Namespace:     flags.addonNamespace,
+		RotationTime:  time.Hour * 24 * 7,
+	}
+	if err := webhookTLS.ManageCertificates(managementFrequency); err != nil {
```

Comment:
What happens if addons-manager restarts ? In the webhook code I don't see the certificate being written to the local filesystem on addons-manager restart.

https://github.com/vmware-tanzu/tanzu-framework/blob/main/pkg/v1/webhooks/webhooks.go#L65

In the case of restart the certificate is valid but its not written to local filesystem. Am i missing something ?

Reply:
on addons-manager start the certificates are installed for the first time in line [308 above ](https://github.com/vmware-tanzu/tanzu-framework/pull/2417/files#diff-b162de15343c757b9843525404e1b18de3623219dee9197c8eca4b6c07fe2de7R308) (call to if _, err := webhooks.InstallNewCertificates..)

everytime the addons-controller binary is restarted new certificates are generated and written to the filesystem. 

@shyaamsn is that what you mean by addons-manager restart?

Reply:
Ah ok got it. Is it a good practice to regenerate certificate every time addons-controller restarts ?

Reply:
It seems the least path of resistance.  In order to validate existing tls secrets from a previous "existance" of an addons-controller we would have to keep track somehow of the status of the addons-controller.. 
And even if we are able to keep track of state (restart, first start, crash, etc..) 
we have no way of assuring the tls certifictes are valid.  or that all the configured webhooks obtained the correct cabundle. 
for example, if the addons-controller was restarted by a crash right in the middle of updating the webhook configurations, we could end up with somewebhook configurations with a valid ca and some with an invalid. 
that situation would not correct itself until the existing (left over) tls certificates become invalid (time expiration). 

The simplest approach, although a bit brute force, is to assume tls certificates are invalid on addons-controller start, and replace with new ones.

but with that said, I think is worth looking at making the certificates in the webhook-tls secret be the source of true. 

we could validate the content of webhook-tls everytime we start and addons-controller. then write them to file if they are valid.

Reply:
refactored code to better handle restart. The new version, looks at the existing webhook tls secret and validatest the certificates there. If valid it will use them and install it (add to tls.crt, key.crt, and webhook configurations)
If not valid, new certificates will be genreated and installed